src/python/CRABInterface/RESTUserWorkflow.py

```diff
@@ -428,6 +428,21 @@ def validate(self, apiobj, method, api, param, safe): #pylint: disable=unused-ar
             else:
                 safe.kwargs["acceleratorparams"] = None
 
+            # Reject it if inputblocks is provided for USER dataset.
+            if param.kwargs.get('inputblocks', None) and \
+               isDatasetUserDataset(safe.kwargs['inputdata'],
+                                    parseDBSInstance(safe.kwargs['dbsurl'])):
+                msg = "'inputblocks' for USER dataset is not supported."
```

Comment:
why do we need this ?

Reply:
To simplify code in TW. Because we never do tape recall for USER dataset.

Reply:
but we still want to allow to process just one block in a USER dataset. Anyhow the decision not to recall USER dataset should better be enforced in only one place, which is very clear and easy to find since it is a soft policy which may change.

Reply:
After I rechecked the code, this [code block](https://github.com/novicecpp/CRABServer/blob/60f29ab3beb851ac3fdbeb2b2748923d5b8403f5/src/python/TaskWorker/Actions/DBSDataDiscovery.py#L453-L478) already handled USER dataset very well. No additional code is needed.

I am removing USER dataset check.